src/classes/smith/smith.ts

```diff
@@ -0,0 +1,123 @@
+import _ from "lodash";
+import { IBehaviour } from "../../types/character/behaviour";
+import { IEquipment } from "../../types/shop/equipment";
+import { IItem } from "../../types/shop/item";
+import { ISmith } from "../../types/character/smith";
+import { IStat } from "../../types/character/stat";
+import { IWeapon } from "../../types/shop/weapon";
+import { SmithTypesEnum } from "../../enums/smith-types.enum";
+import { EquipmentSpotsEnum } from "../../enums/equipment-spots.enum";
+import { ObjectTypesEnum } from "../../enums/object-types.enum";
+import { StatNamesEnum } from "../../enums/stat-names.enum";
+import { SmithBehaviourEnum } from "../../enums/smith-behaviour";
+import { WeaponCategoriesEnum } from "../../enums/weapon-categories.enum";
+import Smiths from "../../data/models/smith-schema";
+import names from "../../data/texts/smiths/names.json";
+
+export class Smith implements ISmith {
+  public behaviour: IBehaviour;
+  public description: string;
+  public equipment?: IEquipment[];
+  public name: string;
+  public items?: IItem[];
+  public startingLevel: number;
+  public stats: IStat[];
+  public type: SmithTypesEnum;
+  public weapons?: IWeapon[];
+
+  constructor() {
+    this.behaviour = this._generateBehaviour();
```

Comment:
The constructor is only useful if you provide overrided values inside it as argument.
You are not doing that here.
So all the code you added inside the constructor could be directly be done when creating the variables above.
It will reduce the class complexity.

Reply:
Ok, I will remove the constructor ^^